addon/MetadataParser.js

```diff
@@ -41,7 +41,7 @@ MetadataParser.prototype = {
     return new Promise(resolve => {
       const doc = this._getDocumentObject(raw);
       resolve(this._formatData(getMetadata(doc, url), url));
-    });
+    }).catch(err => Cu.reportError(err));
```

Comment:
I think it's ok, both ```getMetadata``` and ```_formatData``` always return something, and if either of them throw an exception we'll be able to catch it here. So I *think* it's fine....